server.js

```diff
@@ -64,43 +62,48 @@ wsServer.on('connection', function connection(ws, request) {
-                console.log('ExitRoom Request received from client %o', jsonObj)
-                roomId = util.exitRoom(lobby, jsonObj.roomId, jsonObj.playerId)
-                break
-            case 'GamePlayEvent':
-                var room = lobby.getRoomById(jsonObj.roomId)
-                if (room !== undefined) {
-                    room.broadcastGamePlayEvent(lobby, jsonObj)
-                }
-                break
-            default:
-                console.log('Unknown msg Received from client with type ' + jsonObj.type)
-                break
+        try {
+            jsonObj = JSON.parse(msgStr)
+            switch (jsonObj.type) {
+                case 'JoinOrCreateRoom':
+                    console.log('JoinOrCreateRoom Request received from client %o', jsonObj)
+                    roomId = util.joinOrCreateRoom(lobby, jsonObj.playerId, jsonObj.playerName, jsonObj.playerAvatar, jsonObj.maxPlayersPerRoom)
+                    break
+                case 'GetRooms':
+                    console.log('GetRooms Request received from client %o', jsonObj)
+                    var connection = lobby.getPlayerConnection(playerId)
+                    connection.send(new events.RoomsListEvent(lobby.rooms).convertToJSONString())
+                    break
+                case 'GetAvailableRooms':
+                    console.log('GetAvailableRooms Request received from client %o', jsonObj)
+                    var connection = lobby.getPlayerConnection(playerId)
+                    connection.send(new events.AvailableRoomsListEvent(lobby.availableRooms).convertToJSONString())
+                    break
+                case 'CreateRoom':
+                    console.log('CreateRoom Request received from client %o', jsonObj)
+                    roomId = util.createRoom(lobby, jsonObj.playerId, jsonObj.playerName, jsonObj.playerAvatar, jsonObj.maxPlayersPerRoom)
+                    break
+                case 'JoinRoom':
+                    console.log('JoinRoom Request received from client %o', jsonObj)
+                    roomId = util.joinRoom(lobby, jsonObj.roomId, jsonObj.playerId, jsonObj.playerName, jsonObj.playerAvatar)
+                    break
+                case 'ExitRoom':
+                    console.log('ExitRoom Request received from client %o', jsonObj)
+                    roomId = util.exitRoom(lobby, jsonObj.roomId, jsonObj.playerId)
+                    break
+                case 'GamePlayEvent':
+                    var room = lobby.getRoomById(jsonObj.roomId)
+                    if (room !== undefined) {
+                        room.broadcastGamePlayEvent(lobby, jsonObj)
+                    }
+                    break
+                default:
+                    console.log('Unknown msg Received from client with type ' + jsonObj.type)
+                    break
+            }
+        }
+        catch (err) {
+            console.log('Error parsing received message \n'+err)
```

Comment:
Should we notify the client that the message was bad?

Reply:
Probably OK. The format expectation needed "type" minimally, and the handlers should catch their own exceptions.

There's nothing the client to do besides print the text of the invalid or corrupted message ("cosmic rays" or otherwise causing a parse error.